### DISCUSSION TITLE
GitHub: It is time to listen to your users and either restore the previous searching system or priotize fixing the myriad issues that the community has raised.

### DISCUSSION THREAD

**Discussion by AndrewBuiltThis** [community/community#84737] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

GitHub introduced its new search in May of 2023 and now, nearly a year after implementation, I think the community pretty uniformly agrees: the changes implemented made the product [worse](https://stackdiary.com/githubs-new-code-search-is-bad/). _Far, far_ worse. I'm going to attempt to organize the feedback from previous year's posts, but, ironically it may be incomplete since I am relying on the half-baked search to return results. Here we go: 

- [19-12-2022](https://github.com/orgs/community/discussions/42129): General complaints about search, particularly related to the "recently indexed" filter. u/look responds indicating GitHub wont do it because "it wouldn't be useful" but sorting by commit date _would_ be useful, but they have no plans to do so.
- [26-12-2022]( https://github.com/orgs/community/discussions/42651): Unanswered search bug wherein the search is not returning results.
- [10-1-2023](https://github.com/orgs/community/discussions/43891): Not all instances of matching terms are found. u/look acknowledges this and states that GitHub makes the decision for you by prioritizing the "best" result. 
- [11-1-2023](https://github.com/orgs/community/discussions/43980): undocumented restriction of GitHub search reported wherein if "2 individual lines in [a] file [are] over 4096 bytes [it]... prevents indexing." [Here](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations) are the other code search limitations. To save you a click, here are the publicly stated limitations of search: Vendored and generated code is excluded; Empty files and files over 350 KiB are excluded; Lines over 1,024 characters long are truncated; Only UTF-8 encoded files are included; Very large repositories may not be indexed; Exhaustive search is not supported
- [15-1-2023](https://github.com/orgs/community/discussions/44388): Multiple reports of inconsistent behavior of search, particularly related to the indexing conditions. Multiple users had to use band-aid fixes to "force" reindexing (sharing repo as private, and the back to public, etc.)
- [19-1-2023](https://github.com/orgs/community/discussions/44809): Unanswered bug report related to highlighting behavior within new code search
- [21-1-2023](https://github.com/orgs/community/discussions/44952): Missing ability to exclude all repos that are a fork of a given repo. u/look agrees this is a good idea, but doesn't have a timeline or plan for implementing.
- [27-1-2023](https://github.com/orgs/community/discussions/45538): code search not working for several dozen users. Original post was redirected to use the new search, which was marked as "answered" only to follow by more than a dozen other complaints of the same issue in the new search. 
- [31-1-2023](https://github.com/orgs/community/discussions/45901): Unanswered bug report of missing search results via new search
- [31-1-2023](https://github.com/orgs/community/discussions/45922): Unanswered feedback about code view hijacking page search functionality, the inability to search code that is not in view due to lazy rendering, and limitations of code search relative to browser search. No responses from GitHub.
- [2-2-2023](https://github.com/orgs/community/discussions/46179): Unanswered bug regarding the new search's inability to show large data/files or provide file size indication. Previous functionality was removed in new search. 
- [6-2-2023](https://github.com/orgs/community/discussions/46527): Inability to search with case sensitivity. u/look acknowledges this is on the roadmap, but has not been implemented and no timetable is offered. 
- [8-2-2023](https://github.com/orgs/community/discussions/46794): Unacknowledged accessibility and usability bug reports of the new search on mobile. No responses from GitHub, including after OP followed up three weeks later.
- [9-2-2023](https://github.com/orgs/community/discussions/46906): Old (read: functional) GitHub search reported as changing behavior. This aligns with the continued roll-out of the new search. User is forced to migrate to new beta search.
- [14-2-2023](https://github.com/orgs/community/discussions/47292): Missing wildcard search capabilities. u/look responds with plan to implement, but no updates have been provided on thereafter. 
- [ 21-2-2023](https://github.com/orgs/community/discussions/48124): Unanswered bug related to GitHub's search's ability to handle non-English keywords. OP shows that Google's search finds the result, but GitHub's does not. No responses.
- [ 24-2-2023](https://github.com/orgs/community/discussions/48531): Unanswered feedback related to new search's UI. 
- [ 27-2-2023](https://github.com/orgs/community/discussions/48703): Multiple reports of issues with the search, culminating in u/look informing users that generated files (i.e. yarn.lock) are not supported for searching, nor is time-based sorting, and acknowledges the search is incomplete as a trade-off for limiting the index size.
- [27-2-2023](https://github.com/orgs/community/discussions/48734): Unanswered feedback requesting the code highlighting of search results to favor lines that match all terms. If a user searches for two conditions, the current output will not consistently identify the line that contains both search terms.
- [8-3-2023](https://github.com/orgs/community/discussions/49510): Unresolved request to limit searches to an organization with GitHub cloud. u/look says this is planned, but offers no timetable or subsequent update.
- [13-3-2023](https://github.com/orgs/community/discussions/49907): Another unanswered request to prevent code search from hijacking your browser's search functionality (ctrl-f)
- [15-3-2023](https://github.com/orgs/community/discussions/50171): Unanswered bug report related to inconsistent search behavior when searching in repos.
- [22-3-2023](https://github.com/orgs/community/discussions/50718): Request to keep the old search prior to deprecation due to the already significant reports of issues and missing functionality in the beta. u/colinwm replies to confirm they've tried the new search. No follow ups following OP's confirmation.
- [27-3-2023](https://github.com/orgs/community/discussions/51103): Missing language support in search. OP instructed to submit a "New Language Request" to linguist. 
- [28-3-2023](https://github.com/orgs/community/discussions/51193): Another ignored feedback post wherein users explicitly request GitHub to not deprecate the existing search and to reconsider the rushed implementation. No responses from GitHub. 
- [29-3-2023](https://github.com/orgs/community/discussions/51313): Yet another unanswered request to address code search's hijacking of the browser's search (ctrl-f). No response. 
- [3-4-2023](https://github.com/orgs/community/discussions/51859): Any files that contain any non-UTF-8 characters are excluded from search. u/look's response is that this is a feature, not a bug.
- [3-4-2023](https://github.com/orgs/community/discussions/51889): Yet _another_ unanswered request to address code search's hijacking of keyboard controls. No response.
- [3-4-2023](https://github.com/orgs/community/discussions/51892): Complaint about implementation of line-length limit for search exclusions.
- [4-4-2023](https://github.com/orgs/community/discussions/51957): Request to include fancy-regex for search to better support users. u/look confirms they will not do this because fancy-regex have expensive runtimes.
- [6-4-2023](https://github.com/orgs/community/discussions/52114): Another complaint about GitHub ignoring long lines from search. OP asks if this will be addressed at some point in the future, but is not responded to.
- [9-4-2023](https://github.com/orgs/community/discussions/52404): u/look confirms that newly added repos are not added to index as quickly as updated ones. Interestingly, he suggest searching against the new repo to initiate its indexing. This seems to suggest that items aren't indexed until after they are searched, which is ... concerning?
- [13-4-2023](https://github.com/orgs/community/discussions/52723): Unanswered accessibility and UI request related to code highlighting and contrast. No response from GitHub. 
- [15-4-2023](https://github.com/orgs/community/discussions/52932): Bring back sort:indexed for code searches. u/colinwm says they heard this feedback, but don't have a plan to address it as it is "complex." He states he'll provide an update when they have a plan. He hasn't replied since, ignoring the 90+ calls to at least provide the old search back if they're unable to implement. 
- [17-4-2023](https://github.com/orgs/community/discussions/53041): Rendering and selection mismatch for code search result. u/look confirms this is a bug and is being worked on.
- [25-4-2023](https://github.com/orgs/community/discussions/53703): Unanswered accessibility and UI issue related to unintended flashing/blinking of elements on page.
- [26-4-2023](https://github.com/orgs/community/discussions/53764): Search results are overwhelmingly dated. The time-filter for results was removed. Multiple users requesting the old version to be made available again. No replies to feedback from GitHub team, excepting a response where u/colinwm helps a user with the path: qualifier.
- [28-4-2023](https://github.com/orgs/community/discussions/53954): Unanswered request to restore the ability sort by index time or provide access to the previous API. 
- [1-5-2023](https://github.com/orgs/community/discussions/54192): Request to disabled the new search and restore the previous. OP updates that support informed him this is no longer possible. 
- [2-5-2023](https://github.com/orgs/community/discussions/54278): Multiple reports of the issues related to the UI, performance, and accessibility of the new search. No responses from GitHub.
- [3-5-2023](https://github.com/orgs/community/discussions/54331): Another request to allow for time filtering for search. No responses from GitHub.
- [9-5-2023](https://github.com/orgs/community/discussions/54973): Another request to restore the sort options for code searches. No responses from GitHub.
- [10-5-2023](https://github.com/orgs/community/discussions/55021): Another request to restore the sort options for code searches. No responses from GitHub.
- [10-5-2023](https://github.com/orgs/community/discussions/55065): Unanswered bug report of the new search's inability to handle .zip files when using the path qualifier. No responses from GitHub.
- [10-5-2023](https://github.com/orgs/community/discussions/55084): Unanswered request to restore the ability to sort by date, provide a datetime overview, and existing saved queries now failing to return results. Additionally, OP acknowledges the heavy skew toward returning old results and the absence of new results. No responses from GitHub. 
- [10-5-2023](https://github.com/orgs/community/discussions/55088): Unanswered request to allow users to suppress identical files from being excluded from results. This functionality was present in the old search. No responses from GitHub. 
- [10-5-2023](https://github.com/orgs/community/discussions/55095): Multiple reports of GitHub's new search UI being broken and inaccessible on mobile. UI issues persisted until later October of 2023.
- [11-5-2023](https://github.com/orgs/community/discussions/55178): Unanswered accessibility issues related to GitHub's search bar and their use of qbsearch-input instead of 'input.' No responses from GitHub.
- [11-5-2023](https://github.com/orgs/community/discussions/55202): GitHub will only provide 100 results for code, regardless of the number of matches it finds, u/look confirms this is intended behavior. 
- [12-5-2023](https://github.com/orgs/community/discussions/55322): Multiple user reports of code search being broken and useless. Multiple community responses, echoing the sentiment and requesting the old API be restored. No responses from GitHub. 
- [12-5-2023](https://github.com/orgs/community/discussions/55328): Another post requesting the code search be reverted and expressing continued dissatisfaction that GitHub is ignoring user feedback. No responses from GitHub.
- [12-5-2023](https://github.com/orgs/community/discussions/55339): Another post reporting the 100 result limit as a bug and something that needs to be addressed for the search to be usable. No responses from GitHub.
- [13-5-2023](https://github.com/orgs/community/discussions/55350): User complaint about the inability to sort by time, the randomness of results, and the general degradation of the service. No responses from GitHub.
- [13-5-2023](https://github.com/orgs/community/discussions/55367): Another complaint about .zip files not being searched. u/look confirms that Non-UTF-8 files are not in the index. He further states GitHub cannot promise they will ever restore this. 
- [13-5-2023](https://github.com/orgs/community/discussions/55410): Yet another unanswered feedback post, complaining about the lack of date-sorting, no datetime on overview, and previously valid searches now returning nothing. No responses from GitHub.
- [14-5-2023](https://github.com/orgs/community/discussions/55445): Another complaint about the 100 result limit and expressing frustration that GitHub is not responding to any of these feedback posts. No responses from GitHub.
- [14-5-2023](https://github.com/orgs/community/discussions/55465): Request to restore previous search as new one is riddled with issues. Over two (2) dozen community replies agreeing and reporting issues, and expressing frustration. u/look has taken to manually indexing the organization repos of users who respond to this post. This suggests that there are thousands of similarly un-indexed repos across GitHub that will remain unindexed. Community members consider making petitions to force GitHub to restore the changes. Others consider migrating to other products. 
- [15-5-2023](https://github.com/orgs/community/discussions/55502): Community petition to force GitHub to listen to feedback and undo their breaking changes. Additionally, more frustration at GitHub for just ignoring complaints. Several community members reply in agreement. No responses from GitHub.
- [15-5-2023](https://github.com/orgs/community/discussions/55552): Unanswered complaint about GitHub's 100 result maximum. No responses from GitHub.
- [15-5-2023](https://github.com/orgs/community/discussions/55565): Request to restore the ability to sort code results by date. Multiple community members echo the sentiment. No responses from GitHub.
- [16-5-2023](https://github.com/orgs/community/discussions/55639): Unanswered complaints about the maximum result limit, the inability to sort by time, and the un-findability of previously discoverable resources. No responses from GitHub.
-  [17-5-2023](https://github.com/orgs/community/discussions/55704): Unanswered issue report related to code navigation. OP followed up multiple times, most recently in November to confirm the issue still remains. No responses from GitHub.
- [20-5-2023](https://github.com/orgs/community/discussions/55966): Unanswered post about GitHub no longer returning results for saved searches. No response from GitHub.
- [22-5-2023](https://github.com/orgs/community/discussions/56097): Another request to filter code results by date. u/look says this is no longer possible. He then _pretends_ this is the first time they are hearing about this and asks for details. OP provides and u/look says he will pass this onto the team. 
- [27-5-2023](https://github.com/orgs/community/discussions/56521): Unanswered bug that demonstrates searching for any 8-digit number the search fails with a 502 error. No responses from GitHub.
- [29-5-2023](https://github.com/orgs/community/discussions/56654): Unanswered bug report of modulus character breaking search parameters. u/look replied and said this has been passed on to the team. As of 6-1-2024, this still is an issue and has not been addressed.
- [29-5-2023](https://github.com/orgs/community/discussions/56661): Another user complaining about the new search being broken and not being able find what they need. They lament that previous searches are now returning 95% less results. No responses from GitHub.
- [29-5-2023](https://github.com/orgs/community/discussions/56677): Multiple users reporting issues with the search not finding valid items, the inability to sort by date/recency, and the loss of existing functionality. No responses from GitHub.
- [29-5-2023](https://github.com/orgs/community/discussions/56678): User reports of being rate-limited after a single search. Personally, I have also experienced this. No responses from GitHub.
- [5-6-2023](https://github.com/orgs/community/discussions/57194): User complaints that existing searches are now failing to return results after update. u/look responds that the new search is actually better (but that since their repo contains non-UTF-8 characters it won't be returned).
- [6-6-2023](https://github.com/orgs/community/discussions/57353): wrong file position being accessed in code viewer. u/look confirms this is a bug and indicates there are issues with searching when not in the default branch. Said differently, if you are viewing the non-default branch, the code viewer will select the corresponding text from the default branch. No timeline given on implementing a fix or subsequent update.
- [11-6-2023](https://github.com/orgs/community/discussions/57747): Unanswered accessibility bug with UI in dark mode. No responses from GitHub.
- [13-6-2023](https://github.com/orgs/community/discussions/57747): User reports of repos being indexed inconsistently. Community member responds with assistance. No responses from GitHub.
- [13-6-2023](https://github.com/orgs/community/discussions/57964): Another report of the wrong file position being accessed in code viewer when viewing the non-default branch.
- [17-6-2023](https://github.com/orgs/community/discussions/58306): Another complaint related to the UTF-8 search restriction. 
- [21-6-2023](https://github.com/orgs/community/discussions/58623): Report of "search failed. Please try again later" wherein previously configured GitHub searches are no longer working.
- [28-6-2023](https://github.com/orgs/community/discussions/59556): Another report of different behavior between users related to indexing. Different users get different results. No responses from GitHub.
- [6-7-2023](https://github.com/orgs/community/discussions/60237): Searching code is now restricted behind a login. u/look confirms they implemented this and have no plans to change this.
- [11-7-2023](https://github.com/orgs/community/discussions/60603): Request for filtering code by dates. u/look states this is not possible and there are no plans to implement.
- [12-7-2023](https://github.com/orgs/community/discussions/60702): Results are not being returned by star-count. u/Akash1134 asks for clarification, OP provides clarification on the same day, and then is ghosted. OP follows up in August, but is not responded to. Filtering by stars is no longer available/functional.
- [16-7-2023](https://github.com/orgs/community/discussions/60992): Another request to restore the previous search. No responses from GitHub.
- [6-8-2023](https://github.com/orgs/community/discussions/62926): Another user complaint about GitHub gatekeeping code behind accounts (i.e. having to login to search for code). 
- [10-8-2023](https://github.com/orgs/community/discussions/63335): Yet another complaint about the new search not returning results. u/look states that since the repo contains a folder named "external" it is being flagged as a vendor data and being omitted from the search results. Said differently, **if your project contains a folder named 'external,' it will not be indexed by GitHub.** This suggests a really rudimentary and inaccurate process for flagging content. When called out for this, u/look stops replying. 
- [10-8-2023](https://github.com/orgs/community/discussions/63353): Another user complaint about legitimate results being omitted. This time, it is because the file is larger than 350KB. u/look says they have no plans to increase the file size limit. He also references they are developing tooling so users can see what files are being excluded (which is odd because if a search result can return both an inclusive list and an excluded list, it sounds like you can just return the actual list).
- [11-8-2023](https://github.com/orgs/community/discussions/63484): Another user complaining that their previously discoverable content and searches are no longer working in the current system. No responses from GitHub.
- [12-8-2023](https://github.com/orgs/community/discussions/63542): Inability to search for releases by title. No responses from GitHub.
- [13-8-2023](https://github.com/orgs/community/discussions/63611): Request to sort results by the number of commits. No responses from GitHub.
- [14-8-2023](https://github.com/orgs/community/discussions/63708): Unacknowledged bug report related to saved searches passing the incorrect values to the URL. It looks like it is not properly encoding the whitespaces. No responses from GitHub.
- [18-8-2023](https://github.com/orgs/community/discussions/64172): User identifies that the number of results is inaccurate. u/look states that this number is not actually representative of the results and that, depending on the number of results, the search API handles duplicate files differently for the appearance of more results even though there are not.
- [23-8-2023](https://github.com/orgs/community/discussions/64775): User asks for clarification on the statement "very large repositories may not be indexed." u/look states that "it depends" but its around 50GB and/or 75,000 unique paths, but those two variables interact, so the actual conditions for being considered a "very large repository" remain nebulous. 
- [24-8-2023](https://github.com/orgs/community/discussions/64802): Another user reports this issue of GitHub hijacking your ctrl-f functionality. As usual, no responses from GitHub.
- [6-9-2023](https://github.com/orgs/community/discussions/66105): Another user reports the lack of dates on the overview, the removal of recently updated sorting, and the 100 result (5-page) limit being disastrous. You guessed it, no responses from GitHub.
- [6-9-2023](https://github.com/orgs/community/discussions/66113): Another user cannot find results due to the changes to how GitHub selectively indexes content. u/look confirms that "not all repositories are indexed on creation, but if someone issues a search, we add it." 
- [7-9-2023](https://github.com/orgs/community/discussions/66257): User reports inconsistent and changing results when executing the same queries against the new search when filtering by star-count. No responses from GitHub. 
- [13-9-2023](https://github.com/orgs/community/discussions/67001): A user asks for a timeline for when the arbitrary 100 result limit will be removed. ~GitHub immediately responds~ No responses from GitHub.
- [27-9-2023](https://github.com/orgs/community/discussions/68444): Additional users requesting to revert the search changes and trying to mobilize the community to force GitHub to act. 
- [4-10-2023](https://github.com/orgs/community/discussions/69226): User reporting bug wherein unauthenticated users are provided false results rather than accurate error messages when searching. No responses from GitHub.
- [16-10-2023](https://github.com/orgs/community/discussions/70249): Community letter written to GitHub asking for the recent changes to be reverted as they have severely degraded the product and the consistency of the services. No responses from GitHub.
- [24-10-2023](https://github.com/orgs/community/discussions/71906): User complains that code results no longer indicate how many resources are returned within that particular language. This was previously available in the functional search. No responses from GitHub.
- [14-11-2023](https://github.com/orgs/community/discussions/75515): Users provide feedback of not being able to sort by time, the lack of star-count filter, and the arbitrary UTF-8 indexing requirement. No responses from GitHub.
- [17-11-2023](https://github.com/orgs/community/discussions/75870): Wildcard searches trigger an "exceeded quota" error and return a 429 error. u/look confirms that since this is "too short" of a query, it is inefficient and thus is blocked. 
- [26-11-2023](https://github.com/orgs/community/discussions/77046): User complains about needing to be logged in to search for code. Several community members echo this complaint and lament that this change was co-deployed with the search api's deconstruction (read: "update"). Many boots were licked, many goons were gooning, and GitHub's u/martinwoodward marked his own answer as correct even though it said nothing of use.
- [14-12-2023](https://github.com/orgs/community/discussions/79052): Another complaint about GitHub's arbitrary limit of 100 code results. No responses from GitHub.
- [21-12-2023](https://github.com/orgs/community/discussions/81396):  Another complaint about GitHub's arbitrary limit of 100 code results. No responses from GitHub.
-[ 24-12-2023](https://github.com/orgs/community/discussions/81714): Unanswered bug report wherein archived resources are being returned as "NOT is:archived." No responses from GitHub.

*takes deep breath

I don't understand how Microsoft, GitHub's business daddy, can say that things are too complex or would be too computationally intensive to address. After all, these are the same folks who have no problems bringing your machine to its knees so that Microsoft Teams can run. The fact this has been so universally decried as a leap backwards (and down the stairs (and out the back door (and into the garbage bins))) for over a year with no attempt to meaningfully engage with the community, address the valid concerns, or provide timetables for the few issues that have been "passed on to the team" is scandalous. And the seeming disregard for any accessibility, internationalization, or fair-use concerns seems borderline discriminatory. Perhaps that is the best avenue to take. 

So @martinwoodward, @look, @colinwm, @Akash1134: you're welcome for going through every post and comment in this community for the last year+ to identify all of the posts that still need attention. Please take these feedback seriously as GitHub is not providing the service _we know you are capable of_. How do we know this? Because you _were_ providing the service we needed less than year ago. 

This is a community of pretty clever people and if you really think this is just too difficult to solve, why not let us solve it? Share the source code. If you're unable or unwilling to fix these issues for the community, I have great news: you run a platform where you can share your code/processes _with_ them.  And based on the fervor of the community's response to these changes, they'd be pretty keen on investigating and resolving them because these changes are driving us to your competitors - or to employ "computationally intensive" processes against your system to get what we need. 

I am not terribly confident you'd legitimately consider this, but directly asking the question forces you into the position of either accepting the help or rejecting it (and removes the "no one ever offered" response from the table). And I think the community would rather know where you stand so we can make our decisions. 

If you got all the way through this post, thank you for taking the time to genuinely listen; if not, I know what to expect: _No responses from GitHub._ 

- a

**Comment by AndrewBuiltThis** [comment]
Tagging  @martinwoodward, @look, @colinwm, @Akash1134 to see if they are active with the community.

**Reply by AndrewBuiltThis** [reply]
Tagging @martinwoodward, @look, @colinwm, @Akash1134 to see if they are active with the community.

**Comment by martinwoodward** [comment]
Hey Andrew - thanks for that comprehensive summation.

I know the search team continue to actively invest in making search better on GitHub, in particular semantic code search. I couldn't give you a timeline on when each of the points addressed would be answered, but I think it's fair to say we are committed to the new architecture and pushing it forwards to allow it to work for more and more folks rather than rolling it back.

I hope to see continued progress on some of the issues the community have highlighted and you have diligently summarised above. Appreciate that answer doesn't help you at all - but just wanted to get get back to you so it didn't feel like you were talking into the void even if you were not seeing the direct results you wanted in the timeline you wanted them.

Thanks again.

**Reply by AndrewBuiltThis** [reply]
@martinwoodward, I appreciate the response but am disappointed with the lack of any answer. With no timeline, no specifics or recognition _of _ the points I catalogued for y'all, it sounds like you aren't really interested in the community's feedback or addressing the issues that have been raised. 

I reviewed every post on the topic for the last year, 103 in total, and at least 51 one of them remain unanswered. It is hard to put any value in your response and the commitment of the team to actually serve the community when nearly 50% of all community feedback is completely ignored. Even this post required me to follow-up. 

Does GH share there action plans, roadmaps, or feature backlog? If not, I think that could be an valuable, transparent way to engage with the community so we can make decisions about the platform. Thanks again for responding, as you said, it is nice to know that this isn't just being tossed completely into the void.

**Reply by jsoref** [reply]
As one of the users tagged more than once in items listed here, I sympathize with the poster.

I just did a quick spot check and some of these items have been fixed, some haven't been.

The cost for GitHub to read and respond to them shouldn't be particularly high. If you need more information from a reporter (because they didn't include enough text), you could ask for it (sometimes we, -- one of the examples of mine listed here -- include a wall of links). If we don't respond in the same amount of time as it took for GitHub to respond (plus a month if you're responding promptly...), then you can close the item.

Generally, these feedback items are actionable:
1. "oh, that's a bug", we'll look into fixing it
2. "oh, we should document that constraint", we'll open a ticket to [github/docs: issues](https://github.com/github/docs/issues)
3. "I can't reproduce, can you provide more information?"
4. "oh, we fixed that, can you verify?"
5. "I don't understand, can you clarify?"

Many of these should be addressable with that rubric in under 5 minutes (apiece).

As the vendor with the public feedback channel, you have to be responsible for maintaining the public feedback channel (and this discussion is a good round-up indicting you for failing to do that).

Falling behind by over a year means it'll take some time to catch up, but it's still quite manageable (speaking as someone who has worked on triage for much larger products / issue bases).

**Comment by XDYB** [comment]
There are business opportunities in anger, and new business opportunities are expected.

**Comment by K-O-V-I-D** [comment]
I have created two support tickets, it's has been 15+ days with no response from the support team, actually my account was flagged and I don't even know the reason!! Can anyone let me know where  I can get a sooner reply as my campus placements are starting..please help!!

**Comment by radiomix** [comment]
I stumbled across a _similar_  issue searching in Github Actions UI: **Search only returns the first 100 results**. 

I created this public repo to verify: https://github.com/radiomix/gha-search-in-logs

**Comment by LegalizeAdulthood** [comment]
> After all, these are the same folks who have no problems bringing your machine to its knees so that Microsoft Teams can run.

Minor nit: this isn't the way MS works.  Consider each product team essentially a separate company, so whatever happens on Microsoft Teams, or Microsoft Office, has no bearing on what happens at GitHub.  Most of the poor software we see coming from MS results from the reward mechanisms they have as part of their corporate structure and career ladder.  (My opinion, obviously.  Duh.  Who else's opinion would it be?)

**Comment by darkst0rm223** [comment]
nothing ever happens, they wont fix this shitty excuse of a search engine. maybe someone could make a alternative using the api
